Kubernetes-CLI.adoc

```diff
@@ -0,0 +1,63 @@
+= Kubernetes CLI
+
+The Infinispan CLI currently has two modes: command-line and interactive. 
+To improve the user experience when using the Infinispan Operator we can enhance the CLI to add Kubernetes capabilities.
+
+== Native
+
+Using Quarkus we can build the CLI as a native executable for the three most common platforms: Linux x86_64, OS X, Windows.
+
+== Kubernetes mode
+
+Installing the executable as `kubectl-infinispan` allows invocation as a `kubectl/oc` plugin, e.g.:
+
+`kubectl infinispan ...`
+
+Using the Fabric8 Kubernetes Client we can automatically detect the existence of a Kubernetes configuration (default to `~/.kube/config`) and automatically connect to the Kubernetes cluster.
+
+=== Kubernetes commands
+
+The Infinispan Kubernetes CLI plugin defaults to using the `datagrid` namespace and the `infinispan` service name. These can be overridden in each command using the `--namespace` parameter and the name argument.
+
+==== Install the operator
+
+`kubectl infinispan install [-n namespace]`
+
+Install the Infinispan Operator in the specified namespace. 
+This should also install the relevant OLM subscription and operator group so that the operator can be updated.
+
+==== Create an Infinispan service
+
+`kubectl infinispan create service [-n namespace] [-r replicas] [--expose-type=type] [--expose-port=port] [--expose-host=hostname] [servicename]`
+
+Creates an Infinispan service:
+* `-n namespace` selects the namespace. Defaults to `infinispan-operator`.
+* `-r replicas` selects the initial number of replicas. Defaults to 1.
+* `--expose-type LoadBalancer|NodePort|Route` selects how the service should be exposed.
+* `--expose-port port` specifies the exposed port for LoadBalancer and NodePort types.
+* `--expose-host hostname` specifies the exposed hostname for the Route type.
+
+==== Scale an Infinispan service
+
+`kubectl infinispan scale -r replicas [servicename]`
+
+Scale an Infinispan service to the specified number of replicas
+
+===== Interact with a running Infinispan service
+
+`kubectl shell servicename`
```

Comment:
Is `infinispan` dropped intentionally in here? Won't this require extra kubectl-shell plugin to be developed? Since it's supposed to be infinispan specific I think it should be "namespaced": `kubectl infinispan shell(connect?) servicename`

Reply:
Oversight.